modules/launcher/src/main/java/org/mule/module/launcher/log4j2/LoggerContextConfigurer.java

```diff
@@ -116,26 +117,21 @@ private void disableShutdownHook(LoggerContext context)
     private void configureMonitor(MuleLoggerContext context)
     {
         Configuration configuration = context.getConfiguration();
-        if (!(configuration.getConfigurationMonitor() instanceof FileConfigurationMonitor))
+        File configFile = null;
+        if (context.getConfigFile() != null)
         {
-            File configFile = null;
-            if (context.getConfigFile() != null)
-            {
-                configFile = new File(context.getConfigFile().getPath());
-            }
-            else if (!StringUtils.isEmpty(configuration.getName()))
-            {
-                configFile = new File(configuration.getName());
-            }
+            configFile = new File(context.getConfigFile().getPath());
+        }
+        else if (!StringUtils.isEmpty(configuration.getName()))
+        {
+            configFile = new File(configuration.getName());
+        }
 
-            if (configFile != null && configuration instanceof Reconfigurable)
-            {
-                configuration.setConfigurationMonitor(new FileConfigurationMonitor(
-                        (Reconfigurable) configuration,
-                        configFile,
-                        getListeners(configuration),
-                        DEFAULT_MONITOR_INTERVAL_SECS));
-            }
+        if (configFile != null && configuration instanceof Reconfigurable)
+        {
+            configuration.getWatchManager().setIntervalSeconds(DEFAULT_MONITOR_INTERVAL_SECS);
+            FileWatcher watcher = new ConfiguratonFileWatcher((Reconfigurable) configuration, getListeners(configuration));
+            configuration.getWatchManager().watchFile(configFile, watcher);
```

Comment:
have you empirically tested that this still works? we have no automatic coverage for this